### PR TITLE
Base repo infra + code drop

### LINKED ISSUE #1: Port already-converted lints from the slack-lints repo

### LINKED ISSUE #2: Set up repo infra
- [ ] Build setup
- [ ] Renovate
- [ ] Website
- [ ] Snapshots

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #1
Resolves #2

Ported mostly cleanly from the slackhq/slack-lints repo, with some copied over util code and moving packages around to consolidate on the `com.slack.lint.compose.*` pattern.

Also sets up CI and basic github stuff

### CODE REVIEW

**Comment by ZacSweers** [comment]
Stuff to pay extra attention to
- Maven coordinates stuff (gradle.properties files)
- CI config
- build files

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Gradle Wrapper Validation
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Generate cache key
+        run: ./.github/checksum.sh checksum.txt
+
+      - uses: actions/cache@v3
```
Comment (chrisbanes) [inline .github/workflows/ci.yml:31]: This isn't a blocker. Have you looked at `gradle/gradle-build-action` ([here](https://github.com/gradle/gradle-build-action)). It automates all of the caching so you don't have to do all of this setup manually.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        with:
+          distribution: 'zulu'
+          java-version: '19'
+
+      - name: Build
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: check
+
+      - name: (Fail-only) Bundle the build report
```
Comment (chrisbanes) [inline .github/workflows/ci.yml:49]: Can remove this with the change below

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if: failure()
+        run: find . -type d -name 'reports' | zip -@ -r build-reports.zip
+
+      - name: (Fail-only) Upload the build report
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: error-report
+          path: build-reports.zip
```
Comment (chrisbanes) [inline .github/workflows/ci.yml:58]:
```suggestion
          path: |
            **/build/reports/*
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chrisbanes) [reply .github/workflows/ci.yml:31]: Ah just saw that you do below. Why both?

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    configure<JavaPluginExtension> { toolchain { languageVersion.set(JavaLanguageVersion.of(19)) } }
+
+    tasks.withType<JavaCompile>().configureEach { options.release.set(11) }
+  }
+
+  pluginManager.withPlugin("org.jetbrains.kotlin.jvm") {
+    tasks.withType<KotlinCompile>().configureEach {
+      kotlinOptions {
+        jvmTarget = "11"
+        // TODO re-enable once lint uses Kotlin 1.5
```
Comment (chrisbanes) [inline build.gradle.kts:67]: What kotlin version does it use? 😱

**Review by chrisbanes — APPROVED** [review]
LGTM, nice work on the port!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ZacSweers) [reply .github/workflows/ci.yml:31]: leftover from before that action was ready I think, I can clean this all up a bit

File: build.gradle.kts (hunk 4 above)
Comment (ZacSweers) [reply build.gradle.kts:67]: so this is a good one for #3. In lint 31.1.x it uses kotlin 1.7.10 (this comment is old). I'm not sure what it uses in 31.0.x or 30.4.x

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ZacSweers) [reply .github/workflows/ci.yml]: [7341b9c](https://github.com/slackhq/compose-lints/pull/4/commits/7341b9c6b04c1f8e96c8292f3574e3118cc7595c)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ZacSweers) [reply .github/workflows/ci.yml]: [0bb3909](https://github.com/slackhq/compose-lints/pull/4/commits/0bb39092f6b3c7a70b1db18d27e62d1ddb3951fb)

**Pull request merged**